src/functions/IFunction.java

```diff
@@ -0,0 +1,6 @@
+package functions;
+
+public interface IFunction {
+    void init();
+    void run();
```

Comment:
add overload `Result run(Result state)` which allows for functions to use, record and return calculations

Reply:
May remove run() and assume all functions require keeping track of result state. Functions that don't require can simply ignore and return -- may susceptible to bugs?

Alternatively, require all functions to implement both run() and run(Result state), then functions which don't require can simple do:
```
run() {
  // functionality
}

run(Result state) {
  run()
}
```

More verbose, but less prone to bugs that may arise from accidentally modifying the state